src/platform/plugins/shared/discover/public/application/main/state_management/discover_saved_search_container.ts

```diff
@@ -176,18 +181,38 @@ export function getSavedSearchContainer({
 
   const persist = async (nextSavedSearch: SavedSearch, saveOptions?: SavedObjectSaveOpts) => {
     addLog('[savedSearch] persist', { nextSavedSearch, saveOptions });
+
+    const dataView = nextSavedSearch.searchSource.getField('index');
+    const profileDataViewIds = internalStateContainer.getState().defaultProfileAdHocDataViewIds;
+    let replacementDataView: DataView | undefined;
+
+    // If the Discover session is using a default profile ad hoc data view,
+    // we copy it with a new ID to avoid conflicts with the profile defaults
+    if (dataView?.id && profileDataViewIds.includes(dataView.id)) {
+      replacementDataView = await services.dataViews.create({
+        ...dataView.toSpec(),
+        id: uuidv4(),
+        name: i18n.translate('discover.savedSearch.defaultProfileDataViewCopyName', {
+          defaultMessage: '{name} (copy)',
+          values: { name: dataView.name ?? dataView.getIndexPattern() },
+        }),
```

Comment:
Can we add `managed` flag to this data view (and exclude it when saving a session)?

Reply:
Yes, seems to be fitting here. We could address it separately.

cc @kertal regarding adding "Managed" to data views.

Reply:
I'll see what I can do here in the meantime. While supporting managed data views is likely too much for this PR and 9.0 FF timeframe, I may be able to "mark" it some way similar to ES|QL data views so we can at least differentiate them in the data view picker. I agree with your other comment about seeing "Temporary" by default and would like to fix that if we can.

Reply:
So without fully supporting managed data views, I made it so we can still show specific data views as "Managed" in the picker the same way we handle "Temporary": https://github.com/elastic/kibana/pull/205991/commits/751f36628cff07063bcb1671f775bd54a7b93c5f. I think this would at least make the UX less confusing for users. WDYT?
<img src="https://github.com/user-attachments/assets/b13f1c8c-60c5-4c4e-b0f2-ca00048794c0" width="500" />

I also fixed showing ad hoc data view in the mobile data view picker since it seems that was never supported before.